enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj

```diff
@@ -940,97 +940,100 @@
                        (get "snippet: things")
                        :snippet-id)))))))))
 
-(deftest card-has-parameter-with-source-card-test
-  (let [db        (atom nil)
-        table     (atom nil)
-        field     (atom nil)
-        card      (atom nil)
-        card2     (atom nil)
-        extracted (atom nil)]
-    (testing "Card that has parameter with source card must be deserialization"
-      (ts/with-empty-h2-app-db
-        (reset! db    (ts/create! Database :name "my-db"))
-        (reset! table (ts/create! Table :name "CUSTOMERS" :db_id (:id @db)))
-        (reset! field (ts/create! Field :name "NAME" :table_id (:id @table)))
-        (reset! card  (ts/create! Card :name "source card"))
-        (reset! card2 (ts/create! Card
-                                   :name "the card"
-                                   :parameters [{:id                   "abc"
-                                                 :type                 "category"
-                                                 :name                 "CATEGORY"
-                                                 :values_source_type   "card"
-                                                 ;; card_id is in a different collection with dashboard's collection
-                                                 :values_source_config {:card_id     (:id @card)
-                                                                        :value_field [:field (:id @field) nil]}}]))
+(deftest card-and-dashboard-has-parameter-with-source-is-card-test
```

Comment:
@calherries this is very close to an e2e test because we're testing both export and load here. Think we still need an e2e in yaml_test.clj?

Reply:
Tricky question. Usually this kind of test wouldn't be sufficient because it doesn't test `serdes.base/storage-path` for example, but this function isn't called for `ParameterCard` because its inlined. So the downside of this test vs e2e is it's hard to prove it works without knowing how it's implemented. Even though we know it works now, I would prefer every serialized entity to have an e2e test for this reason.